File: backend/src/datamodels.py
from pydantic import BaseModel
from enum import Enum

class Severity(str, Enum):
    PASS = "pass"
    FLAG = "flag"
    FAIL = "fail"

class Finding(BaseModel):
    section: str
    severity: Severity
    issue: str
    detail: str
    recommendation: str

class MethodologyFlags(BaseModel):
    ite_codes_cited: bool
    peak_hours_defined: bool
    study_area_justified: bool
    los_methodology_stated: bool
    mitigation_proportional: bool

class TIAReview(BaseModel):
    summary: str
    overall_rating: Severity
    findings: list[Finding]
    methodology_flags: MethodologyFlags

File: frontend/app/components/DataModels.ts
export enum Views {
  Homepage,
  Results,
}

export enum Severity {
  Pass,
  Fail,
  Flag,
}

export interface Finding {
  section: string;
  severity: Severity;
  issue: string;
  detail: string;
  recommendation: string;
}
export interface MethodologyFlags {
  ite_codes_cited: boolean;
  peak_hours_defined: boolean;
  study_area_justified: boolean;
  los_methodology_stated: boolean;
  mitigation_proportional: boolean;
}

export interface TIAReview {
  summary: string;
  overall_rating: Severity;
  findings: Finding[];
  methodology_flags: MethodologyFlags;
}
